doc/refactoring_design.md

```diff
@@ -0,0 +1,481 @@
+# Keeper Contract Design for Tethys
+
+***DISCLAIMER: THIS IS A WORK IN PROGRESS***
+
+This document provides technical details and describes the high-level design of keeper contract in ocean protocol.
+
+## 1. Overview
+
+### 1.1 Core Modules
+
+From the top-level point of view, there are total 9 moduels as shown in the below:
+
+<img src="img/refactoring/modules.jpg" width="1000" />
```

Comment:
- Tribes is one of governance models, that ocean could support. We should keep under governance. 
- Exchange and Payment: Exchange/swapping is sub-module in payment (only if the exchange is not Instantaneously, I mean there is no lock/release in the exchange).